### DISCUSSION TITLE
Question about how to get the contact forces and the local center of pressure position

### DISCUSSION THREAD

**Discussion by peterRassolow** [loco-3d/crocoddyl#1316] [Q&A]
Hi,

 Thank you and congratulations for you library!

I was wondering if there were any functions to get the following data that I would like to plot:
- the contact forces related to a 6D contact.
- the position of the local center of pressure (it can be retrieved from the local forces, but maybe a function is already available?).

If so, do you have any example on how to use these functions?

 Thank you!
 Best, 
 Peter

**Comment by cmastalli** [comment]
Hi @peterRassolow,

You can retrieve 6D contact forces and their derivatives if your problem defines 6D contact. You can compute their center of pressure yourself via the `CopSupport` class. 

You can find force information via `action_data.differential.multibody.contacts.contacts`.

I hope this can help you. 

Best,
Carlos

**Comment by peterRassolow** [comment]
Hi @cmastalli 

 Thank you for your prompt answer!
 
 Best,
 Peter
